test/javascripts/turbohead_test.coffee

```diff
@@ -120,3 +133,22 @@ describe 'TurboHead', ->
         newRequest(['a.js', 'b.js']).then ->
           assertScriptCount(2)
           assert.isTrue(requests[1].isFulfilled)
+
+  describe 'script errors', ->
+    it 'sends events for script errors', ->
+      promise = new Promise (resolve) ->
+        $(document).one('page:script-error', (evt) ->
+          assert.equal(evt.originalEvent.data.url, 'a.js')
+          resolve()
+        )
+
+      newRequest(['a.js'])
+      failScriptDownload()
+      promise
+
+    it 'fulfills requests with script errors', ->
+      newRequest(['a.js', 'b.js', 'c.js'])
```

Comment:
Obviously don't want the major refactor, but be careful of relying too much on context for tests (a recipe for leaky/ hard to debug tests), if you guys were the ones that set this up.